src/Form/Control/Upload.php

```diff
@@ -56,19 +55,20 @@ protected function uploaded(array $file): void
 
         $entity->save();
         $this->setFileId($entity->get('token'));
+
+        return null;
     }
 
-    protected function deleted(string $token): JsExpressionable
+    protected function deleted(string $token): ?JsExpressionable
     {
         $model = $this->entityField->getField()->fileModel;
         $entity = $model->loadBy('token', $token);
 
-        $js = new \Atk4\Ui\JsNotify(['content' => $entity->get('meta_filename') . ' has been removed!', 'color' => 'green']);
```

Comment:
this message was misleading, file is deleted after the form is saved only (if not temporary)